frontend/app/(defaultLayout)/not-found.tsx

```diff
@@ -1,13 +1,20 @@
 import { DEFAULT_STORE_CODE } from '@config/env';
 import { getLayoutServerSideProps } from '@layouts/default/server';
 import { NotFoundPage } from './components/not-found-page';
+import { PageFrame } from './components/page-frame';
+import { RegionMenu } from './@regionMenu/components/region-menu';
 
 export default async function NotFound() {
-   const layoutPropsPromise = getLayoutServerSideProps({
+   const layoutData = await getLayoutServerSideProps({
       storeCode: DEFAULT_STORE_CODE,
    });
 
-   const layoutProps = await layoutPropsPromise;
-
-   return <NotFoundPage {...layoutProps} />;
+   return (
+      <PageFrame
+         layoutData={layoutData}
+         regionMenu={<RegionMenu regions={layoutData.stores} />}
+      >
+         <NotFoundPage />
+      </PageFrame>
```

Comment:
This `not-found` component currently serves as the global 404 component and as the 404 component for routes in `(defaultLayout)`. Based on the [component hierarchy](https://nextjs.org/docs/app/building-your-application/routing#component-hierarchy) described in the docs, it seems like this implementation would result in 2 page frames for 404s on routes within `(defaultLayout)`.

```tsx
<Layout>
  <DefaultLayout>
    <PageFrame>
      <ErrorBoundary fallback={
        <NotFound>
          <PageFrame>
            <NotFoundPage />
          </PageFrame>
        </NotFound>
      }>
        <Page />
      </ErrorBoundary>
    </PageFrame>
  </DefaultLayout>
</Layout>
```

Reply:
You're right, but there's a catch. My refactor is not changing the current behaviour (if you check main branch we're rendering the page frame in the not found) and I think the reason is that the version of Next.js we're using now, not found pages are not affected by nested layouts. When we will upgrade Next.js this will behave as you are suggesting (I've tested this during my work on #2144).
I know it's a silly breaking change, but I don't make the rules 😂

Reply:
> the version of Next.js we're using now, not found pages are not affected by nested layouts

Wow that is surprising! Thanks for explaining 👍🏻